### PR TITLE
Update to types

### PR DESCRIPTION
**Pull request by nxtlvlrob**
Made changes to types in two places required for creating a proxy ballot

### CODE REVIEW

File: src/ballotBox.ts [hunk 1]
```diff
@@ -284,7 +283,7 @@ export const castProxyVote = async (request, netConf: EthNetConf) => {
 }
 
 const ProxyProposalInputRT = t.type({
-    ballotSpec: Bytes32RT,
+    ballotSpec: t.string,
```
Comment (XertroV) [inline src/ballotBox.ts]: Why did you change this?

File: src/ballotBox.ts [hunk 2]
```diff
@@ -371,5 +370,26 @@ export const getProxySequenceNumber = async (svNetwork: SvNetwork, ballotId: Byt
 
     // Get and return the sequence number
     const bbFarmContract = new web3.eth.Contract(BBFarmAbi, bbFarmAddress)
+    const votingNetworkDetails = await bbFarmContract.methods.getVotingNetworkDetails()
+    console.log('votingNetworkDetails :', votingNetworkDetails);
```
Comment (XertroV) [inline src/ballotBox.ts]: debugLog

File: src/ballotBox.ts [hunk 3]
```diff
@@ -371,5 +370,26 @@ export const getProxySequenceNumber = async (svNetwork: SvNetwork, ballotId: Byt
...
     return await bbFarmContract.methods.getSequenceNumber(ballotId, voterAddress).call()
 }
+
+/**
+ * Takes the network details string returned by the BBFarm contract and returns network specific values
+ * @param {Bytes32} networkDetails [32b unallocated][32b chainId][32b networkId][160b bbFarm addr on foreign network]
+ *
+ * @returns {object} containing the chainId, networkId and address of the remote BBFarm
+ */
+export const extractNetworkDetails = async (networkDetails: Bytes32) => {
```
Comment (XertroV) [inline src/ballotBox.ts]: probs best to include "foreign" in the name, since that's what we use in the SC - either that or "packed"

maybe call it "explodeForeignNetDetails" or something

File: src/light.ts [hunk 4]
```diff
@@ -28,17 +28,61 @@ const AuctionAbi = require('./smart_contracts/CommAuctionIface.abi.json');
...
+export const initializeSvLight = async (netConf: EthNetConf, opts?: InitSvLightOptsPartial): Promise<SvNetwork> => {
+    const _opts = new InitSvLightOpts(opts, {...netConf, useWebsockets: false})
+
+    const { mkProvider } =
+        _opts.useWebsockets
+            ? { mkProvider: new Web3.providers.WebsocketProvider(_opts.webSocketsProvider) }
+            : { mkProvider: new Web3.providers.HttpProvider(_opts.httpProvider) }
+
+    const web3 = new Web3(mkProvider)
+    console.log('web3 :', web3);
```
Comment (XertroV) [inline src/light.ts]: remove

File: src/light.ts [hunk 5]
```diff
@@ -336,7 +377,8 @@ export const getSingularCleanAbi = (requestedAbiName, methodName) => {
         PaymentsAbi: PaymentsAbi,
         AuxAbi: AuxAbi,
         AuctionAbi: AuctionAbi,
-        ERC20Abi: ERC20Abi
+        ERC20Abi: ERC20Abi,
+        UnsafeEd25519DelegationAbi: UnsafeEd25519DelegationAbi
```
Comment (XertroV) [inline src/light.ts]: cur these lines down (remove duplicated variable names)

File: src/ballotBox.ts (hunk 1 above)
Comment (nxtlvlrob) [reply src/ballotBox.ts]: It's the actual ballot spec - not the ballot spec hash

**Comment by nxtlvlrob** [comment]
@XertroV - All changes updated and tests are now passing

File: src/ballotBox.ts (hunk 1 above)
Comment (XertroV) [reply src/ballotBox.ts]: ahh, coolio
